Need better documentation of build-time dependencies

I read https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#building-kubernetes-on-a-local-osshell-environment but it did not tell me what I need to know.  I started with a "minimal Ubuntu" VM on the IBM Cloud, and it does not include all the needed development tools.  For example, it does not include `make`.  So I installed `build-essential`.

Next, when I ran `make verify`, I discovered that some of the verification tests require Docker and the ability to issue the `docker` command.  So I installed Docker and enabled my user to use the `docker` command (i.e., access the docker socket).

Next, when I ran `make verify`, I got the following failure (among others):
```
Verifying verify-openapi-spec.sh

+++ Running case: verify.openapi-spec 
+++ working dir: /home/mspreitz/code/k8s.io/kubernetes
+++ command: bash "hack/make-rules/../../hack/verify-openapi-spec.sh"
etcd v3.3.10 already installed. To use:
export PATH="/home/mspreitz/code/k8s.io/kubernetes/third_party/etcd:${PATH}"
+++ [0531 01:30:05] Building go targets for linux/amd64:
    cmd/kube-apiserver
jq not found. Please install.
```

I also discovered that `make verify` requires some things to be in $GOPATH, even though I have `GO111MODULE=on`:
```
Verifying verify-typecheck.sh

+++ Running case: verify.typecheck 
+++ working dir: /home/mspreitz/code/k8s.io/kubernetes
+++ command: bash "hack/make-rules/../../hack/verify-typecheck.sh"
type-checking:  linux/amd64, windows/386, darwin/amd64, linux/arm, linux/386, windows/amd64, linux/arm64, linux/ppc64le, linux/s390x, darwin/386
ERROR(all) cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go:27:9: could not import k8s.io/api/core/v1 (type-checking package "k8s.io/api/core/v1" failed (/home/mspreitz/go/pkg/mod/k8s.io/api@v0.0.0-20190111032252-67edc246be36/core/v1/generated.pb.go:226:14: could not import github.com/gogo/protobuf/proto (cannot find package "github.com/gogo/protobuf/proto" in any of:
	/usr/local/go/src/github.com/gogo/protobuf/proto (from $GOROOT)
	/home/mspreitz/go/src/github.com/gogo/protobuf/proto (from $GOPATH))))
ERROR(all) cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go:28:9: could not import k8s.io/apimachinery/pkg/apis/meta/v1 (type-checking package "k8s.io/apimachinery/pkg/apis/meta/v1" failed (/home/mspreitz/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20181127025237-2b1284ed4c93/pkg/apis/meta/v1/controller_ref.go:20:2: could not import k8s.io/apimachinery/pkg/runtime/schema (cannot find package "k8s.io/apimachinery/pkg/runtime/schema" in any of:
	/usr/local/go/src/k8s.io/apimachinery/pkg/runtime/schema (from $GOROOT)
	/home/mspreitz/go/src/k8s.io/apimachinery/pkg/runtime/schema (from $GOPATH))))
```
(and a great many others).

--- a/contributors/devel/development.md
+++ b/contributors/devel/development.md
@@ -115,10 +115,29 @@ environment. Recent Linux distros should work out-of-the-box.
 macOS ships with outdated BSD-based tools. We recommend installing [macOS GNU
 tools].
 
+### make
+
+Kubernetes local build system requires `make` command to be present in your corresponding development platform.
+
+To install `make` command:
+- OS X
+    + `xcode-select --install` (Following command will install CLI Tools to your local development environment)
+- Linux
+    + `sudo apt-get install build-essential` (Following command will install essential commands like `gcc`, `make` etc.)
+
+### Docker
+
+Kubernetes Development requires some of the verification tests which are ran through Docker. Hence, you will need [Docker](https://docs.docker.com/v17.09/engine/installation/) Pre-Installed on your development environment. 
+
 ### rsync
 
 Kubernetes build system requires `rsync` command present in the development
 platform.
+
+### jq
+
+Kube-apiserver requires the `jq` to be installed to successfully build your Local Kubernetes Deployment. 
+You can see installation guide for `jq` over [here](https://stedolan.github.io/jq/download/).
 
 ### Go
 

